src/components/organisms/UiMenu/UiMenu.vue

```diff
@@ -17,11 +17,26 @@
             v-bind="item"
             :name="item.name"
           >
-            <UiText>{{ item.label }}</UiText>
+            <UiCheckbox v-if="isButton" :aria-label="item.label" disabled />
+            <UiText tag="span">{{ item.label }}</UiText>
           </slot>
         </UiMenuItem>
       </template>
     </slot>
+    <!-- @slot Use this slot to replace button -->
+    <slot>
+      <UiMenuItemButton v-if="isButton">
+        <template
+          v-for="(_, name) in itemsToRender"
+          #[name]="data"
+        >
+          <slot
+            v-bind="data"
+            :name="name"
+          />
+        </template>
+      </UiMenuItemButton>
+    </slot>
```

Comment:
:( Sorry but no. This "button" should be a custom UiMenuItem...
![Zrzut ekranu 2024-07-24 o 16 07 14](https://github.com/user-attachments/assets/6f70c99a-8f86-4ec3-a139-8d52e690edb7)
Check how to design and order the elements. 
You have
```
Menu
  Checkbox Menu Item
  Checkbox Menu Item
  Checkbox Menu Item
  ...
  Custom Item Menu
```

Reply:
You don't have to think anything of it.